src/metricscollector/collector/apppoller.go

```diff
@@ -0,0 +1,83 @@
+package collector
+
+import (
+	"metricscollector/cf"
+	"metricscollector/db"
+	"metricscollector/metrics"
+
+	"code.cloudfoundry.org/lager"
+	"time"
+)
+
+type AppPoller struct {
+	logger   lager.Logger
+	appId    string
+	interval time.Duration
+	cfc      cf.CfClient
+	noaa     NoaaConsumer
+	database db.DB
+	ticker   *time.Ticker
+	doneChan chan bool
+}
+
+func NewAppPoller(logger lager.Logger, appId string, interval time.Duration, cfc cf.CfClient, noaa NoaaConsumer, database db.DB) *AppPoller {
+	return &AppPoller{
+		logger:   logger,
+		appId:    appId,
+		interval: interval,
+		cfc:      cfc,
+		noaa:     noaa,
+		database: database,
+		doneChan: make(chan bool),
+	}
+
+}
+
+func (ap *AppPoller) Start() {
+	ap.ticker = time.NewTicker(ap.interval)
+	go ap.startPollMetrics()
+
+	ap.logger.Info("app-poller-started", lager.Data{"appid": ap.appId})
+}
+
+func (ap *AppPoller) Stop() {
+	if ap.ticker != nil {
+		ap.ticker.Stop()
+		ap.doneChan <- true
+	}
+	ap.logger.Info("app-poller-stopped", lager.Data{"appid": ap.appId})
+}
+
+func (ap *AppPoller) startPollMetrics() {
+	ap.pollMetric()
```

Comment:
Here is the change. Move the beginning of the loop to make sure we don't need a cycle to get the metrics polled.

```
	for {
		ap.pollMetric()
		select {
		case <-ap.doneChan:
			return
		case <-ap.ticker.C:
		}
	}

```